modules/fish/prompt.nix

```diff
@@ -6,5 +6,9 @@ let
   };
 in ''
   source ${theme}
-  base16-${config.lib.stylix.colors.slug}
+
+  # See https://github.com/tomyun/base16-fish/issues/7 for why this condition exists
+  if test -n "$base16_theme" && status --is-interactive && test -z "$TMUX"
```

Comment:
Is the `status --is-interactive` condition required for the [Home Manager setup](https://github.com/nix-community/home-manager/blob/release-24.05/modules/programs/fish.nix#L469-L480):

https://github.com/danth/stylix/blob/4faa683a029c72a32942ee118595528b27309b7e/modules/fish/hm.nix#L8

It might be required for the NixOS setup:

https://github.com/danth/stylix/blob/4faa683a029c72a32942ee118595528b27309b7e/modules/fish/nixos.nix#L8

Maybe we should inline `prompt.nix` and implement different versions for `hm.nix` and `nixos.nix`.

Reply:
Another solution would be to switch to [this fork](https://github.com/twiggley/base16-fish) and let upstream handle it. Long-term [tinted-shell](https://github.com/tinted-theming/tinted-shell) seems well-maintained, but might require more work and cause new issues

Reply:
> Maybe we should only inject the `test -z "$TMUX"` condition if users have installed `pkgs.tmux`.

Detecting this is hacky especially since it's possible to run programs via `nix run` without installing. I don't think it's worth adding that extra complexity to save one line of bash.

IMO having the condition in both Home Manager and NixOS is fairly harmless and we can avoid duplicate code by keeping them the same.

Reply:
> Detecting this is hacky especially since it's possible to run programs via `nix run` without installing. I don't think it's worth adding that extra complexity to save one line of bash.

True.

> IMO having the condition in both Home Manager and NixOS is fairly harmless and we can avoid duplicate code by keeping them the same.

Sounds good to me.